miio/integrations/petfooddispenser/device.py

```diff
@@ -0,0 +1,111 @@
+import logging
+from typing import Any, Dict, List
+
+from collections import defaultdict
+import click
+
+from miio.click_common import EnumType, command, format_output
+from miio import Device
+
+
+from .status import PetFoodDispenserStatus
+
+_LOGGER = logging.getLogger(__name__)
+
+MODEL_MMGG_FEEDER_PETFEEDER = "mmgg.feeder.petfeeder"
+
+SUPPORTED_MODELS: List[str] = [MODEL_MMGG_FEEDER_PETFEEDER]
+
+AVAILABLE_PROPERTIES: Dict[str, List[str]] = {
+    MODEL_MMGG_FEEDER_PETFEEDER: [
+        "food_status",
+        "feed_plan",
+        "door_status",
+        "feed_today",
+        "clean_days",
+        "power_status",
+        "dryer_days",
+        "food_portion",
+        "wifi_led",
+        "key_lock",
+        "country_code",
+    ],
+}
+
+class PetFoodDispenser(Device):
+    """Main class representing the Pet Feeder / Smart Pet Food Dispenser. """
+
+    _supported_models = SUPPORTED_MODELS
+
+    @command(
+        default_output=format_output(
+            "",
+            "Power source: {result.power_status}\n"
+            "Food level: {result.food_status}\n"
+            "Automatic feeding: {result.feed_plan}\n"
+            "Food bin lid: {result.door_status}\n"
+            "Dispense button lock: {result.key_lock}\n"
+            "Days until clean: {result.clean_days}\n"
+            "Desiccant life: {result.dryer_days}\n"
+            "WiFi LED: {result.wifi_led}\n",
```

Comment:
It would make sense to avoid using the property names from the protocol itself, but use more developer friendly names (similar to use for these printouts).

Reply:
Ah yeah I was going to ask about that.

Some names I've taken from miot spec for another feeder by mmgg (one that's actually compliant... :angry: ), but obviously some things don't translate overly well and wasn't sure on the preferred approach.

So you're suggesting (for example) "door_status" > "food_bin_lid" or something along those lines?

Reply:
Yes, exactly. It makes sense to separate the property names from what is being internally used, as most of the times they are not clear in their meaning.